### DISCUSSION TITLE
Stack Handling in Loops

### DISCUSSION THREAD

**Discussion by charlesbetros** [jonathanpeppers/dotnes#42] [Ideas]
IL ops like Add don't currently work inside of a loop where the number of loops is not constant or when both operands are not constant.

For example, the current implementation
`Stack.Push(Stack.Pop() + Stack.Pop())`
would push once during transpiling.

Would it be better to store only the operand types in our Stack instead of actual values? This way we know some size types will be popped and a new type pushed, but we wouldn't try to keep track of values.

In the lols sample we have something like:
```
int x = 0;
while (true)
{
  x += 1;
}
```
Since the IL ops are processed once we pop 0, pop 1 and push 1. Then jump and repeat.

**Comment by jonathanpeppers** [comment]
@charlesbetros I haven't had a lot of time to work on this project. Generally, I code all day for work, and lately don't feel like coding any extra for fun... That might change at some point, though!

My memory of the `Stack` above is fuzzy at best, but I would maybe try to get an example like this working first:

```csharp
int x = 0;
x += 1;
x += 1;
x += 1;
x += 1;
```

I wonder if that even works with the current code?

**Reply by charlesbetros** [reply]
Me too. :) (Code all day for work) I saw this project during your talk at Build. Just thinking through options.
